client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HotRodServerStartStopTest.java

```diff
@@ -51,6 +74,32 @@ public void testTouchServer() {
       RemoteCache<Object, Object> remoteCache = remoteCacheManager.getCache();
       remoteCache.put("k", "v");
       assertEquals("v", remoteCache.get("k"));
+
+      CacheTopologyInfo info = remoteCache.getCacheTopologyInfo();
+      Integer prevTop = info.getTopologyId();
+
+      killServers(hotRodServer1, hotRodServer2, hotRodServer3);
+
+      Arrays.stream(managers()).forEach(EmbeddedCacheManager::stop);
+      cacheManagers.clear();
+
+      hotRodServer1 = null;
+      hotRodServer2 = null;
+      hotRodServer3 = null;
+
+      ConfigurationBuilder builder = hotRodCacheConfiguration(
+            getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, false));
+      addClusterEnabledCacheManager(builder);
+
+      hotRodServer1 = HotRodClientTestingUtil.startHotRodServer(manager(0), FIRST_SERVER_PORT, new HotRodServerConfigurationBuilder());
+
+      // data was loss on restart
+      assertNull(remoteCache.get("k1"));
+
+      if (prevTop.equals(remoteCache.getCacheTopologyInfo().getTopologyId())) {
```

Comment:
assertFalse?

Reply:
Sure, I was doing other stuff inside the if before, no reason to have it now.